static/src/javascripts/projects/common/modules/commercial/ticker.js

```diff
@@ -11,62 +14,87 @@ const percentageTotalAsNegative = () => {
+    const counterElement = parentElement.querySelector(
+        '.js-ticker-so-far .js-ticker-count'
     );
 
     if (counterElement && counterElement instanceof HTMLElement) {
-        counterElement.innerHTML = `$${count.toLocaleString()}`;
-        if (count >= total) {
-            counterElement.innerHTML = `$${total.toLocaleString()}`;
+        counterElement.innerHTML = `${getLocalCurrencySymbol()}${count[
+            parentElementSelector
+        ].toLocaleString()}`;
+        if (count[parentElementSelector] >= total) {
+            counterElement.innerHTML = `${getLocalCurrencySymbol()}${total.toLocaleString()}`;
         } else {
-            window.requestAnimationFrame(increaseCounter);
+            window.requestAnimationFrame(() =>
+                increaseCounter(parentElement, parentElementSelector)
+            );
         }
     }
 };
 
-const populateText = () => {
-    const goalElement = document.querySelector(
-        '.epic-ticker__goal .epic-ticker__count'
+const populateText = (parentElement: HTMLElement) => {
+    const goalElement = parentElement.querySelector(
+        '.js-ticker-goal .js-ticker-count'
     );
 
     if (goalElement && goalElement instanceof HTMLElement) {
-        goalElement.innerHTML = `$${goal.toLocaleString()}`;
+        goalElement.innerHTML = `${getLocalCurrencySymbol()}${goal.toLocaleString()}`;
     }
 };
 
-const fetchDataAndAnimate = () => {
-    fetch(
-        'https://interactive.guim.co.uk/docsdata-test/1ySn7Ol2NQLvvSw_eAnVrPuuRnaGOxUmaUs6svtu_irU.json'
-    )
-        .then(resp => resp.json())
-        .then(data => {
-            const showCount = data.sheets.Sheet1[0].showCount === 'TRUE';
-            total = parseInt(data.sheets.Sheet1[0].total, 10);
-            goal = parseInt(data.sheets.Sheet1[0].goal, 10);
+const animate = (parentElementSelector: string) => {
+    const parentElement = document.querySelector(parentElementSelector);
+
+    if (parentElement && parentElement instanceof HTMLElement) {
+        populateText(parentElement);
+        window.setTimeout(() => {
+            count[parentElementSelector] = 0;
+            window.requestAnimationFrame(() =>
+                increaseCounter(parentElement, parentElementSelector)
+            );
+            animateBar(parentElement);
+        }, 500);
+    }
+};
 
-            if (showCount) {
-                populateText();
-                window.setTimeout(() => {
-                    window.requestAnimationFrame(increaseCounter);
-                    animateBar();
-                }, 500);
-            }
-        });
+const dataSuccessfullyFetched = () => total && goal && showCount;
+
+const fetchDataAndAnimate = (parentElementSelector: string) => {
+    if (dataSuccessfullyFetched()) {
+        animate(parentElementSelector);
+    } else {
+        fetch(
+            'https://interactive.guim.co.uk/docsdata-test/1ySn7Ol2NQLvvSw_eAnVrPuuRnaGOxUmaUs6svtu_irU.json'
+        )
```

Comment:
Hi @joelochlann, we're not currently polyfilling `fetch` for older browsers, so we're seeing a lot of JS errors due to this line. Would it be possible to import our `fetch` shim into this module instead ([`lib/fetch`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/lib/fetch.js))

Reply:
Yes will do! Sorry about that. I wonder if there's a way to adjust flow/eslint's list of allowed globals so that this would be caught at build time?

Reply:
fixed in #20854